core/src/main/java/org/elasticsearch/index/shard/TranslogRecoveryPerformer.java

```diff
@@ -152,6 +155,7 @@ public void performRecoveryOperation(Engine engine, Translog.Operation operation
                                     .routing(create.routing()).parent(create.parent()).timestamp(create.timestamp()).ttl(create.ttl()),
                             create.version(), create.versionType().versionTypeForReplicationAndRecovery(), Engine.Operation.Origin.RECOVERY);
                     maybeAddMappingUpdate(engineCreate.type(), engineCreate.parsedDoc().dynamicMappingsUpdate(), engineCreate.id(), allowMappingUpdates);
+                    indexingService.preCreate(engineCreate);
```

Comment:
nope, because before in the `pre*` methods we would parse the query in order to validate that the query was valid. In the `post*UnderLock` methods we would parse the query again and add it to the map of parsed Lucene queries.

Now in the `pre*` methods we parse and extract the terms and adds these terms during the document. So if these methods aren't invoked now we end up not matching percolate queries at all.